core/runtime/runtime.cpp

```diff
@@ -10,6 +10,8 @@ namespace runtime {
 c10::optional<RTDevice> get_most_compatible_device(const RTDevice& target_device) {
   LOG_DEBUG("Target Device: " << target_device);
   auto device_options = find_compatible_devices(target_device);
+  auto current_device = get_current_device();
```

Comment:
This probably has performance implications since the current_device_call has significant perf overhead

Reply:
I thought this line would only run when the engine is being instantiated on a new device and not on every inference call, unless `is_switch_required` is entered, but this PR ensures that would not be the case.

Reply:
Default constructor of `RTDevice` invalid device. Add Optional argument to `get_most_compatible_device` if `RTDevice` is missing then add `get_current_device` call.